pkg/metric/middleware_test.go

```diff
@@ -0,0 +1,71 @@
+package metric
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"gotest.tools/assert"
+)
+
+func TestDurationCode(t *testing.T) {
+
+	registry := prometheus.NewRegistry()
+
+	// mock http handler:
+	httpHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+
+	//mock HistogramVec
+	durationHistogram := promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "http_request_duration_seconds_test",
+			Help: "Duration of my request in seconds.",
+		},
+		[]string{"code"},
+	)
+	registry.MustRegister(durationHistogram)
```

Comment:
@Anxhela21 This can also be removed because the PrometheusMiddleware() will register it. I'm pushing this change.
```suggestion
```